pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java

```diff
@@ -253,67 +284,70 @@ private SuccessResponse uploadSegment(@Nullable String tableName, TableType tabl
 
       // Fetch table name. Try to derive the table name from the parameter and then from segment metadata
       String rawTableName;
-      if (tableName != null && !tableName.isEmpty()) {
+      if (StringUtils.isNotEmpty(tableName)) {
         rawTableName = TableNameBuilder.extractRawTableName(tableName);
-        LOGGER.info("Uploading a segment {} to table: {}, push type {}, (Derived from API parameter)", segmentName,
-            tableName, uploadType);
       } else {
         // TODO: remove this when we completely deprecate the table name from segment metadata
         rawTableName = segmentMetadata.getTableName();
-        LOGGER.info("Uploading a segment {} to table: {}, push type {}, (Derived from segment metadata)", segmentName,
-            tableName, uploadType);
+        LOGGER.warn("Table name is not provided when uploading segment: {} for table: {}", segmentName, rawTableName);
```

Comment:
Make it more explicit like: 
`Table name is not provided when uploading segment: {} for table: {} from segment metadata"`?

Reply:
We log this warning when it is not provided through the query param of the rest request. Filed a follow up PR: #8715